scene/gui/line_edit.cpp

```diff
@@ -452,7 +482,10 @@ void LineEdit::_gui_input(Ref<InputEvent> p_event) {
 				case KEY_HOME: {
 
 					shift_selection_check_pre(k->get_shift());
+					bool should_emit_caret_moved = get_cursor_position() != 0;
```

Comment:
You're using two different approaches for this test depending on the case, either storing the old position as an int, or doing the check beforehand... It's equivalent but there's no reason to not use the same approach everywhere. I think storing the old position is clearer logic-wise than doing the check ahead of time.